app/controllers/api/v1/visualization/expression_controller.rb

```diff
@@ -0,0 +1,65 @@
+module Api
+  module V1
+    module Visualization
+      # API methods for visualizing expression data
+      # does NOT contain methods for editing expression data
+      class ExpressionController < ApiBaseController
+        include Concerns::Authenticator
+        include Concerns::StudyAware
+        include Swagger::Blocks
+
+        before_action :set_current_api_user!
+        before_action :set_study
+        before_action :check_study_view_permission
+
+
+        # Returns the specified expression data for the gene within the given study, optimized for rendering
+        # by the SCP UI.
+        # We agreed that there would be no swagger docs for this endpoint, as it is not intended
+        # to be used other than by the SCP UI, and may change dramatically
+        def show
+          if (!@study.has_expression_data? || !@study.can_visualize_clusters?)
+            render json: {error: "Study #{@study.accession} does not support expression rendering"}, status: 400
+          end
+          data_type = params[:data_type]
+          if (data_type == 'violin')
+            render_violin
+          elsif (data_type == 'annotations')
+            render_annotation_values
+          else
+            render json: {error: "Unknown expression data type: #{data_type}"}, status: 400
+          end
+
+        end
+
+        def render_violin
+          cluster = ClusterRenderingService.get_cluster_group(params, @study)
+          annot_params = ExpressionRenderingService.parse_annotation_legacy_params(params, @study, cluster)
+          annotation = ExpressionRenderingService.get_selected_annotation(@study,
```

Comment:
Agree, though I would like something to indicate that they are visualization-focused, rather than CRUD. ClusterVizService?   or should we name them ClusterService, but put them in a visualization module?

Reply:
Makes sense!  "ClusterService, but put them in a visualization module" strikes me as a bit better of the two options.